staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -226,12 +242,12 @@ func (f *ConfigFlags) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, e
 
 	// retrieve a user-provided value for the "cache-dir"
 	// defaulting to ~/.kube/http-cache if no user-value is given.
-	httpCacheDir := defaultCacheDir
+	httpCacheDir := defaultHTTPCacheDir
 	if f.CacheDir != nil {
 		httpCacheDir = *f.CacheDir
 	}
 
-	discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(homedir.HomeDir(), ".kube", "cache", "discovery"), config.Host)
+	discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(localCacheDir, "cache", "discovery"), config.Host)
```

Comment:
for reviewer: not sure whether we should place the discovery cache dir aside the http cache dir here since flag `--cache-dir` looks like only be used to specify http cache dir.